node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -3297,10 +3625,8 @@ mod tests {
         let mut expected_dest_db = src_db.clone();
         expected_dest_db.add_arbitrary_half_neighbor(dest_node.public_key(), src_node.public_key());
         expected_dest_db
-            .node_by_key_mut(disconnected_node.public_key())
```

Comment:
There's a nontrivial problem here. We need to demonstrate, either in this test or another, that the GossipAcceptor filters out the NodeAddr of incoming Gossip where the NeighborhoodDatabase should not know that NodeAddr. That was the original purpose of the `disconnected_node`. Now the `disconnected_node` cannot serve that purpose because it's being removed altogether by the patch-analysis algorithm.

We need to modify this test so that it has a connected Node that A) is far enough from the receiving Node that it shouldn't have a NodeAddr, and B) has a NodeAddr anyway, or we need to remove the `disconnected_node` from this test (since it serves no purpose now) and add an assertion for this behavior to another test, or we need to write an entirely new test that shows this behavior.